sparta/sparta/functional/Register.hpp

```diff
@@ -1332,6 +1332,12 @@ class RegisterBase : public TreeNode
 private:
     RegisterBits computeWriteMask_(const Definition *def) const
     {
+        // Non-power-of-2 size may cause illegal access,
+        // an exception will be thrown in the constructor.
+        if(!isPowerOf2(def->bytes)) {
+            return RegisterBits(nullptr);
+        }
```

Comment:
Would it be better toI just throw a SpartaException or keep this as it is?

Reply:
The write mask is initialized by Class initializer list, but the legal register size is later checked by the constructor ... 

However an odd size would cause illegal access here.

Reply:
I think you should throw.

Reply:
Agree, that’s much better